end-to-end-tests/tst/test_rest_api_with_authorizers.py

```diff
@@ -0,0 +1,131 @@
+        .created("MyCognitoUserPoolClient", ResourceTypes.COGNITO_USER_POOL_CLIENT)
+        .created("MyFunction", ResourceTypes.LAMBDA_FUNCTION)
+        .created("MyFunctionCognitoPermissionProd", ResourceTypes.LAMBDA_PERMISSION)
+        .created("MyFunctionNonePermissionProd", ResourceTypes.LAMBDA_PERMISSION)
+        .created("MyFunctionLambdaRequestPermissionProd", ResourceTypes.LAMBDA_PERMISSION)
+        .created("MyFunctionLambdaTokenPermissionProd", ResourceTypes.LAMBDA_PERMISSION)
+        .created("MyFunctionIamPermissionProd", ResourceTypes.LAMBDA_PERMISSION)
+        .created("MyFunctionRole", ResourceTypes.IAM_ROLE)
+        .created("MyLambdaAuthFunction", ResourceTypes.LAMBDA_FUNCTION)
+        .created("MyLambdaAuthFunctionRole", ResourceTypes.IAM_ROLE)
+    )
+
+    def tearDown(self):
+        self.delete_stack()
+
+    def test_authorizeres_min(self):
+        input_template = os.path.join(
+            os.getcwd(), "end-to-end-tests/input", "rest_api_with_authorizers", "template.yaml"
+        )
+        capabilities = ["CAPABILITY_IAM"]
+        stack_name = self.make_and_verify_stack(input_template, capabilities, self.rest_api_with_authorizers_resources)
+        stack_resources_list = self.get_stack_resources()
+        stack_outputs = self.get_outputs()
+
+        rest_api_id = self.get_stack_resource_with_resource_type(ResourceTypes.APIGW_RESTAPI).get("PhysicalResourceId")
+        api_gateway_client = boto3.client("apigateway")
+        authorizers = api_gateway_client.get_authorizers(restApiId=rest_api_id).get("items")
+        lambda_authorizer_uri = (
+            "arn:aws:apigateway:"
+            + boto3.session.Session().region_name
+            + ":lambda:path/2015-03-31/functions/"
+            + stack_outputs.get("AuthorizerFunctionArn")
+            + "/invocations"
+        )
+        lambda_token_authorizer = self.get_authorizer_by_name(authorizers, "MyLambdaTokenAuth")
+        self.assertEquals(lambda_token_authorizer.get("type"), "TOKEN", "lambdaTokenAuthorizer: Type must be TOKEN")
+        self.assertEquals(
+            lambda_token_authorizer.get("identitySource"),
+            "method.request.header.Authorization",
+            "lambdaTokenAuthorizer: identity source must be method.request.header.Authorization",
+        )
+        self.assertEquals(
+            lambda_token_authorizer.get("authorizerCredentials"),
+            None,
+            "lambdaTokenAuthorizer: authorizer credentials must not be set",
+        )
+        self.assertEquals(
+            lambda_token_authorizer.get("identityValidationExpression"),
+            None,
+            "lambdaTokenAuthorizer: validation expression must not be set",
+        )
+        self.assertEquals(
+            lambda_token_authorizer.get("authorizerUri"),
+            lambda_authorizer_uri,
+            "lambdaTokenAuthorizer: authorizer URI must be the Lambda Function Authorizer's URI",
+        )
+        self.assertEquals(
```

Comment:
`self.assertEquals` should be `self.assertEqual` (this is how it is done in python3).

Reply:
+1. This should come up as a warning in build, please check if there are any other warnings as well.

Reply:
Initially I had written the code in Python2. Updated the code to work in python3